discord/interaction.go

```diff
@@ -344,9 +344,13 @@ type CommandInteractionOptions []CommandInteractionOption
 // CommandInteraction is an application command interaction that Discord sends
 // to us.
 type CommandInteraction struct {
-	ID       CommandID                 `json:"id"`
-	Name     string                    `json:"name"`
-	Options  CommandInteractionOptions `json:"options"`
+	ID      CommandID                 `json:"id"`
+	Name    string                    `json:"name"`
+	Options CommandInteractionOptions `json:"options"`
+	// 	GuildID is the id of the guild the command is registered to
+	GuildID GuildID `json:"guild_id,omitempty"`
+	// TargetID is the id of the user or message targeted by a user or message command
+	TargetID Snowflake `json:"target_id,omitempty"`
```

Comment:
Is there a better type than Snowflake here? This may be either a message or a user ID

Reply:
Probably not. In my opinion, something like

```go
func (c *CommandInteraction) TargetUserID() discord.UserID
func (c *CommandInteraction) TargetMessageID() discord.MessageID
```

is probably almost as convenient.

Reply:
good idea, done!